lib/erb_lint/linter.rb

```diff
@@ -53,12 +53,40 @@ def run(_processed_source)
       raise NotImplementedError, "must implement ##{__method__}"
     end
 
+    def run_and_update_offense_status(processed_source, enable_inline_config = true)
+      run(processed_source)
+      if @offenses.any? && enable_inline_config
+        update_offense_status(processed_source)
+      end
+    end
+
     def add_offense(source_range, message, context = nil, severity = nil)
       @offenses << Offense.new(self, source_range, message, context, severity)
     end
 
     def clear_offenses
       @offenses = []
     end
+
+    private
+
+    def update_offense_status(processed_source)
+      @offenses.each do |offense|
+        offense_line_range = offense.source_range.line_range
+        offense_lines = source_for_line_range(processed_source, offense_line_range)
+
+        if rule_disable_comment?(offense_lines)
+          offense.disabled = true
+        end
+      end
+    end
+
+    def source_for_line_range(processed_source, line_range)
+      processed_source.source_buffer.source_lines[line_range.first - 1..line_range.last - 1].join
+    end
+
+    def rule_disable_comment?(lines)
+      lines.match(/<%# erblint:disable (?<rules>.*#{self.class.simple_name}).*%>/)
```

Comment:
Loosening this up so that we could support both cases for in an erb template and out would be nice.

```erb
<a></a><%# erblint:disable-line AnchorRule %>
<% 
   button = {
      role: "img" # erblint:disable-line IncorrectRoleRule
   }
%>
```

Reply:
Updated ✅